test_common/KokkosKernels_TestVanilla.hpp

```diff
@@ -0,0 +1,177 @@
+#include "KokkosBatched_Vector.hpp"
+
+namespace Test {
+
+template <class ViewTypeA, class ViewTypeB, class ViewTypeC, class ExecutionSpace>
+struct SharedVanillaGEMM {
+  bool A_t, B_t, A_c, B_c;
+  int C_rows, C_cols, A_cols;
+  ViewTypeA A;
+  ViewTypeB B;
+  ViewTypeC C;
+
+  typedef typename ViewTypeA::value_type ScalarA;
+  typedef typename ViewTypeB::value_type ScalarB;
+  typedef typename ViewTypeC::value_type ScalarC;
+  typedef Kokkos::View<ScalarA*, Kokkos::LayoutStride, typename ViewTypeA::device_type> SubviewTypeA;
+  typedef Kokkos::View<ScalarB*, Kokkos::LayoutStride, typename ViewTypeB::device_type> SubviewTypeB;
+  typedef Kokkos::ArithTraits<ScalarC> APT;
+  typedef typename APT::mag_type mag_type;
+  ScalarA alpha;
+  ScalarC beta;
+
+  KOKKOS_INLINE_FUNCTION
+  void operator()(const typename Kokkos::TeamPolicy<ExecutionSpace>::member_type& team) const {
+    Kokkos::parallel_for(Kokkos::TeamThreadRange(team, C_rows), [&](const int& i) {
+      // Give each kokkos thread a vector of A
+      SubviewTypeA a_vec;
+      if (A_t)
+        a_vec = Kokkos::subview(A, Kokkos::ALL(), i);
+      else
+        a_vec = Kokkos::subview(A, i, Kokkos::ALL());
+
+      // Have all vector lanes perform the dot product
+      Kokkos::parallel_for(Kokkos::ThreadVectorRange(team, C_cols), [&](const int& j) {
+        SubviewTypeB b_vec;
+        if (B_t)
+          b_vec = Kokkos::subview(B, j, Kokkos::ALL());
+        else
+          b_vec = Kokkos::subview(B, Kokkos::ALL(), j);
+        ScalarC ab = ScalarC(0);
+        for (int k = 0; k < A_cols; k++) {
+          auto a = A_c ? APT::conj(a_vec(k)) : a_vec(k);
+          auto b = B_c ? APT::conj(b_vec(k)) : b_vec(k);
+          ab += a * b;
+        }
+        C(i, j) = beta * C(i, j) + alpha * ab;
+      });
+    });
+  }
+};
+// C(i,:,:) = alpha * (A(i,:,:) * B(i,:,:)) + beta * C(i,:,:)
+template <class ViewTypeA, class ViewTypeB, class ViewTypeC, class ExecutionSpace>
+struct Functor_BatchedVanillaGEMM {
+  bool A_t, B_t, A_c, B_c, batch_size_last_dim = false;
+  ViewTypeA A;
+  ViewTypeB B;
+  ViewTypeC C;
+
+  using ScalarA      = typename ViewTypeA::value_type;
+  using ScalarB      = typename ViewTypeB::value_type;
+  using ScalarC      = typename ViewTypeC::value_type;
+  using SubviewTypeA = typename Kokkos::View<ScalarA**, Kokkos::LayoutStride, typename ViewTypeA::device_type>;
+  using SubviewTypeB = typename Kokkos::View<ScalarB**, Kokkos::LayoutStride, typename ViewTypeA::device_type>;
+  using SubviewTypeC = typename Kokkos::View<ScalarC**, Kokkos::LayoutStride, typename ViewTypeA::device_type>;
+
+  ScalarA alpha;
+  ScalarC beta;
+
+  KOKKOS_INLINE_FUNCTION
+  void operator()(const typename Kokkos::TeamPolicy<ExecutionSpace>::member_type& team) const {
+    int i = team.league_rank();
+    SubviewTypeA _A;
+    SubviewTypeB _B;
+    SubviewTypeC _C;
+
+    if (batch_size_last_dim) {
+      _A = Kokkos::subview(A, Kokkos::ALL(), Kokkos::ALL(), i);
```

Comment:
Leading underscore with a capital letter...

Reply:
I'll make a pass on all of these at once to fix in a future PR